buildSrc/src/main/groovy/Docker.groovy

```diff
@@ -269,18 +275,39 @@ class Docker {
         TaskProvider<DockerLogsContainer> containerLogs = project.tasks.register("${taskName}LogsContainer", DockerLogsContainer) { logsContainer ->
             logsContainer.with {
                 containerId.set(dockerContainerName)
+                dependsOn containerFinished
                 onlyIf {
-                    cfg.entrypoint && containerFinished.get().exitCode != 0
+                    cfg.entrypoint && (containerFinished.get().exitCode != 0 || cfg.showLogsOnSuccess)
+                }
+            }
+        }
+
+        if (!cfg.copyOut) {
+            // make a wrap-up task to clean up the task work, wait until things are finished, since we have nothing to copy out
+            return project.tasks.register(taskName) { task ->
+                task.with {
+                    if (cfg.entrypoint) {
+                        dependsOn containerFinished, containerLogs
+                        doLast {
+                            // there was an entrypoint specified, if the command was not successful kill the build once
+                            // we're done copying output
+                            if (containerFinished.get().exitCode != 0) {
+                                throw new GradleException("Command '${cfg.entrypoint.join(' ')}' failed with exit code ${containerFinished.get().exitCode}, check logs for details")
+                            }
+                        }
+                    } else {
+                        dependsOn createContainer
+                    }
+                    finalizedBy removeContainer
```

Comment:
Are tasks, which don't specify inputs or outputs, not cached by default? I'm assuming that's the case, so that this task will never be "up-to-date"?

Reply:
This task doesn't do anything, so it doesn't actually matter if it runs. Instead, it has a doLast to throw in some configurations (and doLast always runs), and it makes sure other tasks have correctly run and are up to date when necessary.

The only purpose of this task is a bit of glue to make sure that everything else ran that is necessary to say "yep, it worked" etc, and clean up the now-used container.